### PR TITLE
SLCORE-1031 Modify label for missing binding on hotspots

### PR DESCRIPTION
**Pull request by nquinquenel**
[SLCORE-1031](https://sonarsource.atlassian.net/browse/SLCORE-1031)

[SLCORE-1031]: https://sonarsource.atlassian.net/browse/SLCORE-1031?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: medium-tests/src/test/java/mediumtest/hotspots/HotspotLocalDetectionSupportMediumTests.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ void it_should_not_support_local_detection_in_standalone_mode() {
 
     assertThat(checkResponse)
       .extracting(CheckLocalDetectionSupportedResponse::isSupported, CheckLocalDetectionSupportedResponse::getReason)
-      .containsExactly(false, "The project is not bound, please bind it to SonarQube 9.9+ or SonarCloud");
+      .containsExactly(false, "The project is not bound, please bind it to SonarQube (Server, Cloud)");
```
Comment (thahnen) [inline medium-tests/src/test/java/mediumtest/hotspots/HotspotLocalDetectionSupportMediumTests.java:87]: I've read the branding guide but reading this feels super bad.

File: backend/core/src/main/java/org/sonarsource/sonarlint/core/hotspot/HotspotService.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
 public class HotspotService {
 
   private static final SonarLintLogger LOG = SonarLintLogger.get();
-  private static final String NO_BINDING_REASON = "The project is not bound, please bind it to SonarQube 9.9+ or SonarCloud";
+  private static final String NO_BINDING_REASON = "The project is not bound, please bind it to SonarQube (Server, Cloud)";
```
Comment (thahnen) [inline backend/core/src/main/java/org/sonarsource/sonarlint/core/hotspot/HotspotService.java:53]: Shouldn't we mention here the dependency on SonarQube Server 9.9+ at least?

**Review by thahnen — CHANGES_REQUESTED** [review]
Not sure about the phrasing here, it reads pretty badly.

File: backend/core/src/main/java/org/sonarsource/sonarlint/core/hotspot/HotspotService.java (hunk 2 above)
Comment (nquinquenel) [reply backend/core/src/main/java/org/sonarsource/sonarlint/core/hotspot/HotspotService.java:53]: We only support SQ 9.9 as minimal version now, so this is not relevant anymore

File: medium-tests/src/test/java/mediumtest/hotspots/HotspotLocalDetectionSupportMediumTests.java (hunk 1 above)
Comment (sophio-japharidze-sonarsource) [reply medium-tests/src/test/java/mediumtest/hotspots/HotspotLocalDetectionSupportMediumTests.java:87]: I agree, maybe here we could afford to spell it out `SonarQube Server or SonarQube Cloud`. But not a blocker from my perspective.

**Review by sophio-japharidze-sonarsource — APPROVED** [review]

**Review by thahnen — APPROVED** [review]
LGTM

**Pull request merged**